### ISSUE TITLE
Content-Type parsing issue

### ISSUE THREAD

**Issue by migolovanov** [fastify/fastify#4445]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the regression has not already been reported

### Last working version

3.29.3

### Stopped working in version

3.29.4

### Node.js version

16.18.1

### Operating system

Linux

### Operating system version (i.e. 20.04, 11.3, 10)

22.04

### 💥 Regression Report

The following code
`fastify.addContentTypeParser('application/msgpack', ...`
does not consider `Content-Type: application/msgpack, charset=utf-8` as valid one and raises error `Unsupported Media Type`, which I believe is incorrect.

### Steps to Reproduce

`fastify.addContentTypeParser('application/msgpack', ...`

### Expected Behavior

Content-type charset should not be part of content-type validation

**Comment by climba03003** [comment]
I don't think it is a valid `content-type`.
`application/msgpack; charset=utf-8` would be a valid one.

**Comment by migolovanov** [comment]
Yes, I know. That's probably is the client issue, but it was working fine in 3.29.3 and not working in 3.29.4.

**Comment by migolovanov** [comment]
Sure, no problem. I created this issue just in case someone will meet similar errors after upgrade.

**Comment by climba03003** [comment]
As a workaround, you may use RegExp `/^application\/msgpack(.*);?/` which will do a full text comparison.

**Comment by climba03003** [comment]
cc @fastify/core 
Maybe we need switch from `strict parsing` to `.indexOf() === 0` in `v3` and `v4` instead of `v5`.

Better step to minimize impact for RegExp.

1. Remove `content-type` parsing.
2. Switch `text` comparison to `.indexOf() === 0`
3. Inspect `regexp` and warn if they start with proper `^` or check with `;?`. Since it will expose to the same attack.
4. Add `Security Notice` in document and provide some example for the better `RegExp` usage.

**Comment by jsumners** [comment]
> I don't think it is a valid `content-type`. `application/msgpack; charset=utf-8` would be a valid one. Note the difference between `,` and `;`.

Incorrect. These two content types are equivalent: `application/msgpack` and `application/msgpack; charset=utf-8`.

https://www.rfc-editor.org/rfc/rfc2046.html#section-4.1.2
> A critical parameter that **may** be specified in the Content-Type field

(emphasis added)

The spec path is https://www.rfc-editor.org/rfc/rfc9110.html#name-content-type => https://www.rfc-editor.org/rfc/rfc9110.html#name-media-type => https://www.rfc-editor.org/rfc/rfc2046.html#section-4.1.2

**Comment by climba03003** [comment]
> Incorrect

Please read the comment carefully. 
The issue mention `application/msgpack, charset=utf-8` which is an invalid one.

I mention `application/msgpack; charset=utf-8` in later for clarify, the client should send with `;` as separator instead of `,`

**Comment by jsumners** [comment]
Ah, yes, in that case this is unsupported and not an error we should handle. If a service needs to handle such invalid cases then they can pass their own content type parser:

https://www.fastify.io/docs/latest/Reference/Server/#addcontenttypeparser
```js
'use strict'

const fastify = require('fastify')({ log: { level: 'debug' } })

fastify.addContentTypeParser(
  'text/plain, charset=utf-8',
  { asString: true },
  fastify.defaultTextParser
)

fastify.route({
  path: '/foo',
  method: 'POST',
  async handler (req, reply) {
    let result = ''
    for await (const line of req.body) {
      // Note: it was unexpected that `req.body` is a readable stream
      // this feels like a bug
      result += line
    }

    return reply.send({
      recieved: result
    })
  }
})

fastify.listen({ port: 3000 })
```

**Comment by mcollina** [comment]
@climba03003 would you mind opening the PR for strict parsing? Then we can decide if we ship in now or laterz

**Comment by climba03003** [comment]
@mcollina Here it is. #4450

**Comment by Fdawgs** [comment]
Fixed by #4450?
